SS2-Project/Assets/Starstorm2/Modules/Pickups/Equipments/AffixEthereal.cs

```diff
@@ -1,20 +1,198 @@
-﻿using RoR2;
+﻿using MonoMod.Cil;
+using MSU;
+using RoR2;
+using RoR2.ContentManagement;
+using System.Collections;
 using System.Collections.Generic;
+using UnityEngine;
+
+using RoR2;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Networking;
+using Mono.Cecil.Cil;
+using MonoMod.Cil;
+using System;
+using RoR2.Navigation;
+using RoR2.Projectile;
+using static MSU.BaseBuffBehaviour;
+
 namespace SS2.Equipments
 {
-    //[DisabledContent]
-    /*public sealed class AffixEthereal : SS2EliteEquipment
+#if DEBUG
+    public sealed class AffixEthereal : SS2EliteEquipment
     {
-        public override EquipmentDef EquipmentDef { get; } = SS2Assets.LoadAsset<EquipmentDef>("AffixEthereal", SS2Bundle.Equipments);
+        //public override EquipmentDef EquipmentDef { get; } = SS2Assets.LoadAsset<EquipmentDef>("AffixEthereal", SS2Bundle.Equipments);
+
+        //public override List<MSEliteDef> EliteDefs { get; } = new List<MSEliteDef>
+        //{
+        //    SS2Assets.LoadAsset<MSEliteDef>("edEthereal", SS2Bundle.Equipments),
+        //};
+
+        //public override bool FireAction(EquipmentSlot slot)
+        //{
+        //    return false;
+        //}
+        public override List<EliteDef> EliteDefs => _eliteDefs;
+        private List<EliteDef> _eliteDefs;
+
+        public override NullableRef<List<GameObject>> ItemDisplayPrefabs => null;
 
-        public override List<MSEliteDef> EliteDefs { get; } = new List<MSEliteDef>
+        public override EquipmentDef EquipmentDef => _equipmentDef;
+        private EquipmentDef _equipmentDef;
+
+        public  BuffDef _buffEthereal; //TODO MSU 2.0 replace { get; } = SS2Assets.LoadAsset<BuffDef>("bdEthereal", SS2Bundle.Equipments);
+        public Material _matEtherealOverlay; //=> SS2Assets.LoadAsset<Material>("matEtherealOverlay", SS2Bundle.Equipments);
+
+
+        public override bool Execute(EquipmentSlot slot)
         {
-            SS2Assets.LoadAsset<MSEliteDef>("edEthereal", SS2Bundle.Equipments),
-        };
+            return false;
+        }
 
-        public override bool FireAction(EquipmentSlot slot)
+        public override void Initialize()
+        {
+            // Add relevant hooks
+            IL.RoR2.HealthComponent.TakeDamage += EtherealDeathIL;
+
+            // Add buff overlays
+            BuffOverlays.AddBuffOverlay(_buffEthereal, _matEtherealOverlay);
```

Comment:
EliteEquipments take care of loading the buff and the material overlays by utilizing the metadata found in the ExtendedEliteDef, this line is unecesary.